### PR TITLE
ENH: Use specimen identifiers for pick lists

### PR DESCRIPTION
**Pull request by ElDeveloper**
When a LibraryPrepShotgunProcess or a NormalizationProcess produce a
pick list, use specimen identifiers if these are present.

### CODE REVIEW

**Comment by ElDeveloper** [comment]
Note for reviewers on GitHub: make sure you expand the diff for `labman/db/tests/test_process.py`. GitHub is collapsing the view because of one *really* long line.

File: labman/db/tests/test_process.py [hunk 1]
```diff
@@ -699,6 +700,36 @@ def test_generate_echo_picklist(self):
         obs = NormalizationProcess(2).generate_echo_picklist()
         self.assertEqual(obs, NORM_PROCESS_PICKLIST)
 
+    def test_generate_echo_picklist_with_specimen_id(self):
+        # create the expected data by replacing the sample names
+        lines = [line for line in NORM_PROCESS_PICKLIST.split('\n')]
```
[the remark this answers is not in the document]
Comment (ElDeveloper) [reply labman/db/tests/test_process.py]: I chatted with @wasade offline. Since this is not used anywhere else, I won't add a new function.

**Comment by wasade** [comment]
I didn't realize this was pending, sorry. I'm now +1

**Comment by AmandaBirmingham** [comment]
Can you fill me in on the need for this change?  What is being replaced with specimen id here is not the sample id but the "content", which includes plate and well information (which the wet lab has requested).  The wet lab approved the existing format for the picklists (without this PR) and I'm loathe to go back and ask them to sign off on changes they didn't requests ... unless of course they did request them and I am confused :)

**Comment by ElDeveloper** [comment]
@AmandaBirmingham this came as an addition to the changes from #348. When I talked to Rodolfo he suggested that using the specimen identifier would be handy to see in the pick lists for situations when they troubleshoot something. Rationale being that you want to see the same names here as you see in the plating UI.

**Comment by wasade** [comment]
@AmandaBirmingham, this PR seems to be in purgatory. Given @ElDeveloper, do you believe this should be merged? 

@ElDeveloper, would you be able to resolve conflicts or close based on the response?

**Comment by AmandaBirmingham** [comment]
I understand that this is part of the #348  work.  However, I don't feel the issue discussion to date has addressed my concern, which is that we are replacing a *fully qualified* description of the well contents (which includes not only the sample id but also the plate and well it came from) with a *minimal* description that includes only the specimen id (which is roughly equivalent to the sample id but doesn't include plate and well info).  If the dev team is sure that the wet lab wants specimen_id and does NOT want the provenance plate and well info anymore, then I won't stand in the way of this being merged, but I'd be happier if they put that in writing :D

**Comment by wasade** [comment]
I agree, it would seem that retaining the plate and well is a :+1:. @ElDeveloper, would it be possible to include that when dumping the specimen ID here?

**Comment by antgonza** [comment]
@ElDeveloper ping!

**Comment by ElDeveloper** [comment]
Alright, this has been updated as requested. I ended up making a few more changes. Some of the tests' expected strings were saved as very long lines in the test_process.py file, and it was cumbersome to update these. I have added a "data" folder with these data in there. By removing these lines from the file I also removed the `noqa` directive for flake8, uncovering a number of non PEP8 lines of code.

**Comment by ElDeveloper** [comment]
@AmandaBirmingham  I have gotten a confirmation from Rodolfo, via e-mail:

> I just checked and the [proposed] format will work with the Echo. I did
> notice, however, that the Echo doesn't automatically recognize the column
> header 'Sample'. It expects instead either 'Sample Name' or 'Sample ID' as
> default column headers for sample identification.

Therefore, I have also replaced the header name to be Sample ID.

**Comment by wasade** [comment]
@ElDeveloper, is this good for re-review?

**Comment by ElDeveloper** [comment]
@wasade yes it is 👍

**Comment by ElDeveloper** [comment]
Friendly ping for this to get merged.

**Comment by AmandaBirmingham** [comment]
>  I have gotten a confirmation from Rodolfo, via e-mail:
> 
> > I just checked and the [proposed] format will work with the Echo. I did
> > notice, however, that the Echo doesn't automatically recognize the column
> > header 'Sample'. It expects instead either 'Sample Name' or 'Sample ID' as
> > default column headers for sample identification.

I .... just ... then why did both the "known good" echo files we were provided as specifications have "Sample" as the column header? Oh, my head :neutral_face:

**Comment by ElDeveloper** [comment]
Thanks @AmandaBirmingham!

**Pull request merged**